ee/clickhouse/migrations/0025_remove_dollar_from_materialized_columns.py

```diff
@@ -0,0 +1,48 @@
+import structlog
+from clickhouse_driver.errors import ServerException
+from infi.clickhouse_orm import migrations
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.materialized_columns.columns import get_materialized_columns, materialized_column_name
+from posthog.constants import GROUP_TYPES_LIMIT
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_REPLICATION
+
+logger = structlog.get_logger(__name__)
+
+
+def rename_column(table, current_name, new_name):
+    if CLICKHOUSE_REPLICATION and table == "events":
+        sync_execute(
+            f"ALTER TABLE {table} RENAME COLUMN \"{current_name}\" TO {new_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'"
+        )
+        sync_execute(
+            f"ALTER TABLE sharded_{table} RENAME COLUMN \"{current_name}\" TO {new_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'"
+        )
+    else:
+        sync_execute(f'ALTER TABLE {table} RENAME COLUMN "{current_name}" TO {new_name}')
+    logger.info("Renamed column containing $ sign.", table=table, current_name=current_name, new_name=new_name)
+
+
+def rename_column_if_exists(table, current_name, new_name):
+    try:
+        rename_column(table, current_name, new_name)
+    except ServerException as err:
+        logger.info("Failed to rename column, ignoring.", table=table, current_name=current_name, new_name=new_name)
+
+
+def rename_materialized_columns_with_dollars(database):
```

Comment:
What happens while this is running, or if we fail part way through for some reason? If old code is still trying to access $group will that be an issue?

Reply:
Good question.

We won't fail part way through, but old code will be running even after this is complete. Due to caching a few queries will fail as this is deployed as a result until the new pods are live. As such, I'd deploy this during a quieter timeframe.